src/python/pants/backend/jvm/tasks/classpath_products.py

```diff
@@ -192,16 +192,21 @@ def add_for_target(self, target, classpath_elements):
     """
     self._add_elements_for_target(target, self._wrap_path_elements(classpath_elements))
 
-  def add_jars_for_targets(self, targets, conf, resolved_jars):
+  def add_jars_for_targets(self, targets, conf, resolved_jars_and_directory_digests):
     """Adds jar classpath elements to the products of the provided targets.
 
     The resolved jars are added in a way that works with excludes.
+    :param targets: The targets to add the jars for.
+    :param conf: The configuration.
+    :param resolved_jars_and_directory_digests: A list of tuples of ResolvedJars
```

Comment:
Would it be easier to add a digest to `ResolverJar` instead?

Reply:
I think adding a digest to `ResolvedJar` makes sense too. `ResolvedJar` links coordinates to locations of resolved artifacts. A digest is a sort of location, so it makes sense for `ResolvedJar` to have one.